### PR TITLE
Tox devdeps_npy2

### PR DESCRIPTION
**Pull request by jhunkeler**
@mcara Did this get rebased out of #265?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -27,14 +28,17 @@ changedir = .tmp/{envname}
...
     devdeps: with the latest developer version of key dependencies
+    devdeps_npy2: with numpy 2.x
     cov: and test coverage
 
 deps =
     # The devdeps factor is intended to be used to install the latest
     # developer version or nightly wheel of key dependencies.
-    devdeps: pyerfa>=0.0.dev0
-    devdeps: numpy>=0.0.dev0
+    devdeps: numpy<=2.0.0.dev0
```
Comment (pllim) [inline tox.ini:37]: What is the point of this?

Comment (jhunkeler) [reply tox.ini:37]: When numpy 2 is released both would pick up numpy 2. Oh... That's supposed to be a `<`

Comment (pllim) [reply tox.ini:37]: I think what you want is oldestdeps pin to `numpy==1.*`?

**Review by pllim — CHANGES_REQUESTED** [review]
I think having 2 flavors of numpy is too confusing. Why not use oldestdeps to pin numpy 1.x ? Am I misunderstanding your intention here?

See https://github.com/astropy/astropy/blob/main/tox.ini

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

**Review by mcara — APPROVED** [review]

**Pull request merged**